_docs/Qabel-Protocol-Box.md

```diff
@@ -0,0 +1,578 @@
+meta: null,
+key: "b43feebe528a56bb4f21ef3a8a617714aee2cabc0708c1702a98915ae852ad06",
+ref: "0846C7C6-77F1-11E5-B21E-9CFF64691233",
+},
+{ name: "barfoo.txt", size: 4568, mtime: 1445432120,
+meta: "a9c6ce30-418b-e292-83bc-769a8c72f600",
+key: "042a77edb0d527816ddb3e74457d92e69302099881b9a3181a514696c0fc39bf",
+ref: "8f5da4db-02ab-ca96-1824-3ba8d18a85be"
+}],
+folders: [
+{ name: "some folder",
+key: "fgah28991273814c9123987124f009893043ef75a0dbf3f4eba4a98eaa9b4e6a",
+ref: "aa8c3f39-edc5-00b0-ab8b-ba66d05b60db"
+}],
+externals: [
+{ name: "external share",
+owner: "feffe9928665731c6d6a8f9467308308feffe9928665731c6d6a8f9467308308",
+key: "d570b7fcf9eda9daa648d5ec18ae04x9bd1f7b8d6acbd8764844df5aaae0ff91",
+url: "https://other_bucket.s3.amazonaws.com/users/a3fdc333-a143-85aa-edbf-43adf3ff7315/b6e78ecb-176d-031c-d1d4-eed608ae6e12"
+}]
+}
+```
+
+### Directory Metadata [DM]
+
+
+```
+{
+root: STR, // URL of the VOLUME
+name: "index", // name of the file itself
+last_change_by: UUID, // ID of the device that made the last change
+spec_version: INT,  // version of the VOLUME spec
+					// increment if migrations are needed
+					// and/or the files are incompatible between versions
+version: {version: KEY, time: LONG}, // metadata version and time of change, time should not be trusted
+shared: // description of all shares
+{ shares* },
+files: // list of files in this folder
+[ file* ]
+folders: // list of folders in this folder
+[ folder* ]
+externals: // list of external shares in this folder
+[ external* ]
+}
+```
+
+Note that folders that are not "index" do not have the "shared"-key, as all information about shares in a VOLUME are stored in "index".
+
+The version is a SHA-256 hash built with the following rule:
+```
+version(0) = SHA-256(0x00 || device-id)
+version(n) = SHA-256(0x01 || version(n-1) || device-id)
+```
+
+The device-id is unique for each client and a 128-bit value.
+
+### Shares
+
+The index is the path to the metadata file of the share.
+
+```
+{
+read: [KEY] // List of public keys of identities that this folder is read only shared with
+},
+```
+
+### File Metadata [FM]
+
+A file only has its own FM, if it is shared in a single file share.
+The FM is stored in the path referenced as "meta" in the file object and encrypted
+with a new symmetric key
+
+```
+{
+name: STR, // filename
+spec_version: INT,  // version of the VOLUME spec
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+}
+```
+
+### Objects
+
+File:
+
+```
+{
+name: STR, // object name,
+size: LONG, // uncompressed file size
+mtime: LONG, // modification time as seconds since epoch
+meta: {ref: STR, // path to the FM, if it exists
+       key: KEY} // symmetric key for the FM
+key: KEY, // symmetric key for the block
+block: STR // path to the block without the prefix \<root\>/blocks/
+},
+```
+
+Folder:
```

Comment:
Please add an "aka directory" here, because this spec sometimes uses the term directory or directory object.